src/dmqproto/node/neo/request/Consume.d

```diff
@@ -1004,3 +1004,469 @@ public abstract scope class ConsumeProtocol_v2
+
+    private RequestOnConn.EventDispatcher.MessageParser parser;
+
+    /***************************************************************************
+
+        If true, dataReady() (called when a record was pushed) resumes the
+        fiber.
+
+    ***************************************************************************/
+
+    private bool resume_fiber_on_push;
+
+    /***************************************************************************
+
+        If true, flushBatch() (called when the flush timer expires) resumes the
+        fiber.
+
+    ***************************************************************************/
+
+    private bool resume_fiber_on_flush;
+
+    /***************************************************************************
+
+        Aquired buffer in which the most recently popped value is stored.
+
+    ***************************************************************************/
+
+    private void[]* value_buffer;
+
+    /***************************************************************************
+
+        Aquired buffer in which the batch of records to send is stored.
+
+    ***************************************************************************/
+
+    private void[]* record_batch;
+
+    /***************************************************************************
+
+        Request handler. Reads the initial request args and starts the request
+        main loop.
+
+        Params:
+            connection = connection to client
+            msg_payload = initial message read from client to begin the request
+                (the request code and version are assumed to be extracted)
+
+    ***************************************************************************/
+
+    final public void handle ( RequestOnConn connection, Const!(void)[] msg_payload )
+    {
+        this.connection = connection;
+        this.ed = connection.event_dispatcher;
+        this.parser = this.ed.message_parser;
+
+        cstring channel_name;
+        cstring subscriber_name;
+        this.parser.parseBody(msg_payload, channel_name, subscriber_name);
+
+        if ( !this.prepareChannel(channel_name, subscriber_name) )
+        {
+            this.ed.send(
+                ( ed.Payload payload )
+                {
+                    payload.addConstant(RequestStatusCode.Error);
+                }
+            );
+            return;
+        }
+
+        this.ed.send(
+            ( ed.Payload payload )
+            {
+                payload.addConstant(RequestStatusCode.Started);
+            }
+        );
+
+        this.value_buffer = this.resources.getVoidBuffer();
+        this.record_batch = this.resources.getVoidBuffer();
+
+        try
+        {
+            this.run();
+            return;
+        }
+        catch (ChannelRemovedException e)
+        {
+            // Send the ChannelRemoved message below outside the catch clause to
```

Comment:
An alternative approach to channel-removed handling might be to automatically recreate it here. The client would then never need to know anything about it.

Reply:
If feasible, this would seem to be the cleanest approach.

Reply:
A kind of "RemoveChannel? NO :grumpy:" approach ;)

Reply:
Though *actually*, [this approach](https://github.com/sociomantic-tsunami/dhtproto/issues/39) is probably the best way to go.

Reply:
I agree. It wouldn't make sense to allow removing a channel that has consumers because `RemoveChannel` is meant to be used to undo the mistake of accidentally creating a channel that no one reads from.

Reply:
Right. We should discuss how to make this change to RemoveChannel. Seems like it'll simplify a lot of neo requests.